<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Nicholas Hathaway" />


<title>SeekDeep processClusters Usage</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="commonStyle.css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SeekDeep</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-sitemap"></span>
     
    Installing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="installingSeekDeep_Mac.html">
        <span class="fa fa-apple"></span>
         
        Mac OS
      </a>
    </li>
    <li>
      <a href="installingSeekDeep_Ubuntu.html">
        <span class="fa fa-linux"></span>
         
        Ubuntu
      </a>
    </li>
    <li>
      <a href="installingSeekDeep_Windows.html">
        <span class="fa fa-windows"></span>
         
        Windows
      </a>
    </li>
    <li>
      <a href="installingSeekDeep_vagrant.html">Vagrant/virtual image (any system)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Code
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://github.com/bailey-lab/SeekDeep">
        <span class="fa fa-github"></span>
         
        Github
      </a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-wrench"></span>
     
    Usages
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Pipeline</li>
    <li>
      <a href="extractor_usage.html">extractor/extractorPairedEnd</a>
    </li>
    <li>
      <a href="makeSampleDirectories_usage.html">makeSampleDirectories</a>
    </li>
    <li>
      <a href="qluster_usage.html">qluster</a>
    </li>
    <li>
      <a href="processClusters_usage.html">processClusters</a>
    </li>
    <li>
      <a href="popClusteringViewer_usage.html">popClusteringViewer</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Pipeline Wrapper</li>
    <li class="divider"></li>
    <li class="dropdown-header">Utilities</li>
    <li>
      <a href="genTargetInfoFromGenomes_usage.html">genTargetInfoFromGenomes</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-terminal"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Single End (454/IonTorrent/Stitched Illumina)</li>
    <li>
      <a href="tutorial_IonTorrent_MIDs.html">Ion Torrent with MIDs</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Illumina Paired End</li>
    <li>
      <a href="tutorial_PairedEnd_noMIDs.html">Paired End No MIDs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc Info
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="illumina_paired_info.html">Illumina Paired End Info</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SeekDeep processClusters Usage</h1>
<h4 class="author"><em>Nicholas Hathaway</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#processclusterss-purpose">processClustersâ€™s Purpose</a></li>
<li><a href="#getting-usage-command-line">Getting usage command line</a></li>
<li><a href="#input">Input</a></li>
<li><a href="#output-files">Output files</a><ul>
<li><a href="#column-explaination-for-selectedclustersinfo.tab.txt">Column explaination for selectedClustersInfo.tab.txt</a></li>
<li><a href="#column-explaination-for-populationcluster.tab.txt">Column explaination for populationCluster.tab.txt</a></li>
</ul></li>
<li><a href="#examples">Examples</a><ul>
<li><a href="#running">Running</a></li>
<li><a href="#input-results-file-format">Input results file format</a><ul>
<li><a href="#fastq">Fastq</a></li>
<li><a href="#fasta">Fasta</a></li>
</ul></li>
<li><a href="#pars">Pars</a><ul>
<li><a href="#just-do-a-simple-match-allowing-no-differences">Just do a simple match allowing no differences</a></li>
<li><a href="#allowing-only-1-high-quality-mismatch">Allowing only 1 high quality mismatch</a></li>
<li><a href="#allowing-a-few-low-quality-mismtaches-and-some-1-base-indels-good-for-454-and-ion-torrent">Allowing a few low quality mismtaches and some 1 base indels (good for 454 and Ion Torrent)</a></li>
<li><a href="#allowing-a-few-low-quality-mismtaches-and-but-no-indels-illumina">Allowing a few low quality mismtaches and but no indels (illumina)</a></li>
<li><a href="#allowing-a-few-low-quality-mismtaches-and-but-no-indels-illumina-and-1-high-quality-mismatch">Allowing a few low quality mismtaches and but no indels (illumina) and 1 high quality mismatch</a></li>
</ul></li>
<li><a href="#different-population-parameters">Different Population Parameters</a></li>
<li><a href="#supplying-grouping-info">Supplying grouping info</a></li>
<li><a href="#supplying-expected-reference-sequences">Supplying expected reference sequences</a></li>
<li><a href="#supplying-experiment-name">Supplying experiment name</a></li>
<li><a href="#giving-a-relative-abundance-cut-off">Giving a relative abundance cut off</a></li>
<li><a href="#removing-singletons">Removing singletons</a></li>
<li><a href="#marking-chimeras-and-checking-chimeras">Marking Chimeras and checking Chimeras</a></li>
<li><a href="#extra-cleanup">Extra cleanup</a></li>
</ul></li>
</ul>
</div>

<section id="processclusterss-purpose" class="level1 purposeDiv">
<h1>processClustersâ€™s Purpose</h1>
<p><a href="#TOC">back to top</a><br />
processClusters takes care of several of the final steps of the SeekDeep pipeline.</p>
<ul>
<li><strong>replicate comparison</strong> - If multiple replicates are available per sample it is able to take advantage of this and take only clusters that appear in all replicates<br />
</li>
<li><strong>final results filtering</strong> - A filter is applied to the final results where low frequency clusters and singlet clusters are removed. This step is applied to get rid of artifact sequence that couldnâ€™t be handled by previous extraction and clustering steps and removal of clusters marked as chimeric.<br />
</li>
<li><strong>population comparison</strong> - Final population comparison of all sample final results is done to analyze haplotypes across samples to get haplotypes of the whole population</li>
</ul>
</section>
<section id="getting-usage-command-line" class="level1">
<h1>Getting usage command line</h1>
<p><a href="#TOC">back to top</a><br />
Just typing the name of the program will give a help message on running the program</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters </a></code></pre></div>
<pre><code>SeekDeep processClusters
Ran from inside directory tree set up such that currentDir/SampleDir/RunDir/seqFiles
                             Required Options
                                Clustering
                              --par ParametersFileName; required; default=; ([38;5;202mstd::string)
                          Reading Sequence Input
                            --fasta Input sequence filename, only need 1, fasta text file; required; default=; ([38;5;202mboost::filesystem::path)
                          --fastagz Input sequence filename, only need 1, fasta gzipped file; required; default=; ([38;5;202mboost::filesystem::path)
                            --fastq Input sequence filename, only need 1, fastq text file; required; default=; ([38;5;202mboost::filesystem::path)
                          --fastqgz Input sequence filename, only need 1, fastq gzipped file; required; default=; ([38;5;202mboost::filesystem::path)
                             Optional Options
                            Additional Output
                            --extra Extra Output; default=false; ([38;5;202mbool)
           --writeExcludedOriginals Write out the excluded Originals; default=false; ([38;5;202mbool)
                                Alignment
...
...</code></pre>
<p>Also calling <code>-help</code> will do the same</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --help </a></code></pre></div>
<p>Also all flags in SeekDeep are case insensitive and so all the following would have the same results</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --help </a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="ex">SeekDeep</span> processClusters --HELP</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="ex">SeekDeep</span> processClusters --HeLP</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="ex">SeekDeep</span> processClusters --HeLp</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="ex">SeekDeep</span> PROCESSCLUSTERS --HeLP</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="ex">SeekDeep</span> processclusters --HeLp</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="ex">SeekDeep</span> ProcessClusters --HeLp</a></code></pre></div>
</section>
<section id="input" class="level1">
<h1>Input</h1>
<p><a href="#TOC">back to top</a><br />
The input to processClusters is done by running it in a directory with several results files for each sample in subdirectories below the top directory. An example directory tree below</p>
<pre><code>extraFiles/sampleClusteringDir
â”œâ”€â”€ Sample01
â”‚Â Â  â”œâ”€â”€ MID01
â”‚Â Â  â”‚Â Â  â””â”€â”€ output.fastq
â”‚Â Â  â””â”€â”€ MID02
â”‚Â Â      â””â”€â”€ output.fastq
â”œâ”€â”€ Sample02
â”‚Â Â  â”œâ”€â”€ MID03
â”‚Â Â  â”‚Â Â  â””â”€â”€ output.fastq
â”‚Â Â  â””â”€â”€ MID04
â”‚Â Â      â””â”€â”€ output.fastq
â”œâ”€â”€ Sample03
â”‚Â Â  â”œâ”€â”€ MID05
â”‚Â Â  â”‚Â Â  â””â”€â”€ output.fastq
â”‚Â Â  â””â”€â”€ MID06
â”‚Â Â      â””â”€â”€ output.fastq
â””â”€â”€ Sample04
    â”œâ”€â”€ MID07
    â”‚Â Â  â””â”€â”€ output.fastq
    â””â”€â”€ MID08
        â””â”€â”€ output.fastq</code></pre>
<p>Each output file needs to be named the same (here output.fastq) and the sequences in the output should be named so that they have a suffix indicating how many reads are associated with each cluster. Suffix should be _[NUM] or _t[NUM] where [NUM] is the number of reads associated with each cluster.</p>
</section>
<section id="output-files" class="level1">
<h1>Output files</h1>
<p><a href="#TOC">back to top</a><br />
An output directory will be created to store the output of processClusters, the name of which can be changed by using the <code>-dout</code> flag. In this directory will be several files and directories.</p>
<ul>
<li><strong>selectedClustersInfo.tab.txt</strong> - This is a very table containing all information of replicate comparison, final samples haplotypes, and any population information if population clustering has been done</li>
<li><strong>final</strong> - This directory will contain a sequence file for each sample that was in the clustering and will be the final haplotypes for the analysis. Each sequence in the sequence file will have a suffix in the format _f[NUM] where [NUM] is a relative abundance in the sample so _f1.0 would be 100% of the sample and _f.50 would be a haplotype that took up about 50% of the sample.</li>
<li><strong>sampRepAgreementInfo</strong> - This folder contains several files that report how well replicates agree including a file containing graphs to show this If population clustering was enabled</li>
<li><strong>population</strong> - This directory will be create if population clustering is enable and will contain information about the population of haplotypes found in the samples</li>
<li><strong>population/populationCluster.tab.txt</strong> - This will contain information about how many samples the population haplotypes appear in.<br />
</li>
<li><strong>population/PopUID.fastq</strong> - This will contain the sequences for all the haplotypes found in the population, they will have have a suffix with _f[NUM] where [NUM] is the average relative abundance it was found at. For example if the haplotype was found in 3 different sample at relative abundances of .5, .6, and .7 the suffix here would be _f0.6 If groupings info was supplied</li>
<li><strong>groups</strong> - This directory will be created if grouping information is supplied, see <a href="#supplying-grouping-info">below</a> for details.</li>
</ul>
<section id="column-explaination-for-selectedclustersinfo.tab.txt" class="level2">
<h2>Column explaination for selectedClustersInfo.tab.txt</h2>
<p><a href="#TOC">back to top</a></p>
<p>All the data from the analysis done by processClusters is located in this file. It is a very large file and can be overwhelming at first but all data is conveniently located in one place. How the table is organized is each row represents a haplotype from a sample. Therefore each row will report stats on the haplotype that will have to do with how it was found in the replicates, what the final stats of the haplotype after replicate comparison was, and if population analysis is being done stats on how the haplotype appears in the population. At the same time general stats for each sample and replicate are reported and therefore several columns will have redundant information because there will be several haplotypes for each replicate and sample. Also to define some terms to make sense of the column heading. Each sample will have a one or more replicates, these are normally PCR replicates. Reads from these replicates will first be clustered separately and then compared to each other to make the final data for the sample. The results from this comparison will results in what will be called clusters and each cluster will have a relative abundance in the sample totaling up to 1. After clusters are finalized in the samples they are then the final clusters from each samples are compared to each other to create population data incorporating all samples. This population analysis will result will what will be called haplotypes. So while in the sample these sequences are referred to as clusters and in the population as haplotype to distinguish when talking about each one. Each column has a prefix to indicate what kind of information it being reported on. s_ indicates data about the sample, p_ indicates data about the whole population h_ indicates data about the haplotype, c_ indicates data about the cluster while in the sample and RXX indicates information in the replicates where XX is the replicate number.</p>
<ul>
<li><strong>s_Sample</strong> - The name of the sample the haplotype appears in<br />
</li>
<li><strong>h_popUID</strong> - The population id given to haplotype. These are named with PopUID.XX where XX is the population rank number and this number is determined by the number of samples the haplotype appears in</li>
<li><strong>p_TotalPopulationSampCnt</strong> - This the the total number of samples in the population analysis<br />
<strong><em>Haplotype Info</em></strong></li>
<li><strong>h_PopFrac</strong> - This is the sum of the relative abundances the haplotype was found in divided by the total number of samples in the population.<br />
</li>
<li><strong>h_SumOfAllFracs</strong> - This is the sum of all the relative abundances the haplotype was found at</li>
<li><strong>h_AvgFracFoundAt</strong> - This is the average relative abundance the haplotype was found at<br />
</li>
<li><strong>h_ReadFrac</strong> - This is the total number of reads that are associated with the haplotype across all samples divided by the total number of reads in the the whole experiment</li>
<li><strong>h_SampCnt</strong> - This is the number of samples the haplotype was found in.<br />
</li>
<li><strong>h_SampFrac</strong> - This is the number of samples the haplotype was found in divided by the number of samples in the experiment</li>
<li><strong>h_ReadCnt</strong> - This is the total number of reads associated with this haplotype</li>
<li><strong>h_ClusterCnt</strong> - This is the total number of clusters that went into making this population haplotype, this is normally the number of samples the haplotype was found in because otherwise that would mean a sample contributed more than one cluster to the haplotype.<br />
</li>
<li><strong>h_clusterNames</strong> - This is the names of the contributing clusters<br />
<strong><em>Sample Info</em></strong></li>
<li><strong>s_Name</strong> - The name of the sample the haplotype appears in</li>
<li><strong>s_ReadCntTotUsed</strong> - This is the number of reads used out of this sample given with percentage of the total number of reads given to processClusters, the other percentage would be what was filtered off (low frequency and chimeras)</li>
<li><strong>s_InputClusterCnt</strong> - The total number of clusters at the beginning of the analysis for this sample</li>
<li><strong>s_FinalClusterCnt</strong> - The total number of clusters after analysis<br />
<strong><em>Cluster Info</em></strong></li>
<li><strong>c_clusterID</strong> - The id given to the cluster in the sample, the lower the number the higher the relative abundance in the sample</li>
<li><strong>c_AveragedFrac</strong> - The averaged relative abundance between all replicates used in the analysis (this basically an un-weighted average of the abundances since it wouldnâ€™t depend on read counts)</li>
<li><strong>c_ReadFrac</strong> - The number of reads associated with this cluster divided by the total number of reads found in all replicates, if there is only one replicate this number will be the same as c_AveragedFrac.<br />
</li>
<li><strong>c_ReadCnt</strong> - The total number of reads associated with this cluster summed over all replicates</li>
<li><strong>c_RepCnt</strong> - The number of reps this cluster was found in</li>
<li><strong>c_Consensus</strong> - The consensus sequence for this cluster</li>
<li><strong>c_InputCnt</strong> - The number of contributing replicate clusters to this sample cluster</li>
<li><strong>c_ChiReadCnt</strong> - The number of reads for the replicate clusters that are marked chimeric</li>
<li><strong>c_ChiClusCnt</strong> - The number of replicate clusters that are marked as chimeric</li>
<li><strong>c_ChiRepCnt</strong> - The number of replicated this current sample was marked as chimeric</li>
<li><strong>c_InputNames</strong> - The names of the replicate clusters that contributed to making this sample cluster</li>
<li><strong>RXX_name</strong> - The number of the replicate</li>
<li><strong>RXX_totalClusterCntExcluded</strong> - The total number of replicate clusters that were excluded from this replicate</li>
<li><strong>RXX_totalCntExcluded</strong> - The total number of reads that were excluded from the replicate</li>
<li><strong>RXX_totalFracExcluded</strong> - The fraction of reads excluded from this replicate</li>
<li><strong>RXX_clusterCntChiExcluded</strong> - The number of clusters that were marked chimeric in this replicate</li>
<li><strong>RXX_cntChiExcluded</strong> - The total read count marked as chimeric in this replicate</li>
<li><strong>RXX_fracChiExcluded</strong> - The fraction of the replicate that was excluded due to chimeric formation</li>
<li><strong>RXX_MapFrac</strong> - The relative abundance of this replicate cluster in this replicate</li>
<li><strong>RXX_ReadCnt</strong> - The total number of reads this replicate cluster has</li>
<li><strong>RXX_ClusCnt</strong> - The total number of clusters in this replicate</li>
<li><strong>RXX_totalReadCnt</strong> - The total number of reads in this replicate</li>
<li><strong>c_bestExpected</strong> - A string with the name of the closest reference sequence see <a href="#supplying-expected-reference-sequences">below</a> to see how to compare against expected references</li>
</ul>
</section>
<section id="column-explaination-for-populationcluster.tab.txt" class="level2">
<h2>Column explaination for populationCluster.tab.txt</h2>
<p><a href="#TOC">back to top</a></p>
<ul>
<li><strong>h_popUID</strong> - The population id given to haplotype. These are named with PopUID.XX where XX is the population rank number and this number is determined by the number of samples the haplotype appears in<br />
</li>
<li><strong>p_TotalInputReadCnt</strong> - The total number of reads in the whole experiment/population</li>
<li><strong>p_TotalInputClusterCnt</strong> - The total number of clusters in the whole experiment/population</li>
<li><strong>p_TotalPopulationSampCnt</strong> - The total number of samples in the whole experiment/population</li>
<li><strong>p_TotalHaplotypes</strong> - The total number of final haplotypes in the whole experiment/population</li>
<li><strong>p_meanCoi</strong> - The mean complexity of infection (COI) in the whole experiment/population</li>
<li><strong>p_medianCoi</strong> - The median complexity of infection (COI) in the whole experiment/population</li>
<li><strong>p_minCoi</strong> - The minimum complexity of infection (COI) in the whole experiment/population</li>
<li><strong>p_maxCoi</strong> - The max complexity of infection (COI) in the whole experiment/population</li>
<li><strong>h_PopFrac</strong> - This is the sum of the relative abundances the haplotype was found in divided by the total number of samples in the population.<br />
</li>
<li><strong>h_SumOfAllFracs</strong> - This is the sum of all the relative abundances the haplotype was found at</li>
<li><strong>h_AvgFracFoundAt</strong> - This is the average relative abundance the haplotype was found at<br />
</li>
<li><strong>h_ReadFrac</strong> - This is the total number of reads that are associated with the haplotype across all samples divided by the total number of reads in the the whole experiment</li>
<li><strong>h_SampCnt</strong> - This is the number of samples the haplotype was found in.<br />
</li>
<li><strong>h_SampFrac</strong> - This is the number of samples the haplotype was found in divided by the number of samples in the experiment</li>
<li><strong>h_ReadCnt</strong> - This is the total number of reads associated with this haplotype</li>
<li><strong>h_ClusterCnt</strong> - This is the total number of clusters that went into making this population haplotype, this is normally the number of samples the haplotype was found in because otherwise that would mean a sample contributed more than one cluster to the haplotype.<br />
</li>
<li><strong>h_clusterNames</strong> - This is the names of the contributing clusters<br />
</li>
<li><strong>h_Consensus</strong> - The consensus sequence of all contributing clusters to this haplotype</li>
<li><strong>h_Protein</strong> - An estimated protein, will only be correct if the reading frame starts at 0</li>
<li><strong>c_bestExpected</strong> - A string with the name of the closest reference sequence see <a href="#supplying-expected-reference-sequences">below</a> to see how to compare against expected references</li>
</ul>
</section>
</section>
<section id="examples" class="level1">
<h1>Examples</h1>
<p><a href="#TOC">back to top</a></p>
<section id="running" class="level2">
<h2>Running</h2>
<p><a href="#TOC">back to top</a><br />
processClusters needs only two options to run, name of the results files given by <code>--fastq</code> or <code>--fasta</code> in the directory tree (see <a href="#input">above</a>) and a parameters file <code>--par</code> see <a href="qluster_usage.html##format-of-paramters-files">qluster usage</a> for explanation of parameters file. Alternately to the <code>--par</code> you can use either <code>--noErrors</code> to allow for now errors to collapse between comparisons or <code>--strictErrors</code> to allow for just a few low quality mismatches.</p>
</section>
<section id="input-results-file-format" class="level2">
<h2>Input results file format</h2>
<p><a href="#TOC">back to top</a><br />
Input reads can be fasta or fastq</p>
<section id="fastq" class="level3">
<h3>Fastq</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt</a></code></pre></div>
</section>
<section id="fasta" class="level3">
<h3>Fasta</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fasta output.fasta --par pars.txt</a></code></pre></div>
</section>
</section>
<section id="pars" class="level2">
<h2>Pars</h2>
<section id="just-do-a-simple-match-allowing-no-differences" class="level3">
<h3>Just do a simple match allowing no differences</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --noErrors</a></code></pre></div>
</section>
<section id="allowing-only-1-high-quality-mismatch" class="level3">
<h3>Allowing only 1 high quality mismatch</h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --noErrors --hq 1</a></code></pre></div>
</section>
<section id="allowing-a-few-low-quality-mismtaches-and-some-1-base-indels-good-for-454-and-ion-torrent" class="level3">
<h3>Allowing a few low quality mismtaches and some 1 base indels (good for 454 and Ion Torrent)</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --strictErrors</a></code></pre></div>
</section>
<section id="allowing-a-few-low-quality-mismtaches-and-but-no-indels-illumina" class="level3">
<h3>Allowing a few low quality mismtaches and but no indels (illumina)</h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --strictErrors --illumina</a></code></pre></div>
</section>
<section id="allowing-a-few-low-quality-mismtaches-and-but-no-indels-illumina-and-1-high-quality-mismatch" class="level3">
<h3>Allowing a few low quality mismtaches and but no indels (illumina) and 1 high quality mismatch</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --strictErrors --illumina --hq 1</a></code></pre></div>
</section>
</section>
<section id="different-population-parameters" class="level2">
<h2>Different Population Parameters</h2>
<p>If you want to give different parameters to do the population clustering with just use the â€™-popPar` flag</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --popPar popPars.txt</a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --strictErrors --pop-noErrors</a></code></pre></div>
</section>
<section id="supplying-grouping-info" class="level2">
<h2>Supplying grouping info</h2>
<p><a href="#TOC">back to top</a><br />
To output additional files of final results organized by groups, supply a grouping tab delimited file where the first column has sample names and each additional column is named for a group with sub-groups that each sample belongs to. See below for an example file</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="fu">cat</span> exampleGroups.tab.txt </a></code></pre></div>
<pre><code>samples   age    location
Sample01  child  loc1
Sample02  child  loc2
Sample03  adult  loc1
Sample04  adult  loc2</code></pre>
<p>Just give this file name to the <code>--groups</code> flag</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --groups exampleGroups.tab.txt</a></code></pre></div>
<p>This will create a directory called <strong>groups</strong> in the processClusters output directory. This will contain a series of sub directories of results files split by group and sub-group with a file called sampFile.tab.txt (set up like <a href="#column-explaination-for-selectedclustersinfo.tab.txt">selectedClusters.tab.txt</a>) and a file called popFile.tab.txt (set up like <a href="#column-explaination-forcluster.tab.txt">populationCluster.tab.txt</a>) but will only contain information about that subgroup, the population UIDs will be the same so they can be linked to the main results data. This is still a new feature and will be developed further in future release when analysis between groups will automatically be done but for now the most interesting thing to look at would be the COI stats in each popFile.tab.txt between sub-groups in each group.</p>
</section>
<section id="supplying-expected-reference-sequences" class="level2">
<h2>Supplying expected reference sequences</h2>
<p>To supply a reference file to compare sequences to (this is helpful when you have controls with expected mixture done along with your experiment) use the <code>--ref</code> or <code>--refFastq</code> flag. The <code>--ref</code> flag is for when the sequences are in a fasta file and <code>--refFastq</code> is for when the seq file is a fastq file.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --ref references.fasta</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#or fastq</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --refFastq references.fastq</a></code></pre></div>
</section>
<section id="supplying-experiment-name" class="level2">
<h2>Supplying experiment name</h2>
<p><a href="#TOC">back to top</a><br />
By default the final haplotypes are given the PopUID.XX but you can change the PopUID to be the name of your experiment</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --experimentName myExperiment</a></code></pre></div>
</section>
<section id="giving-a-relative-abundance-cut-off" class="level2">
<h2>Giving a relative abundance cut off</h2>
<p><a href="#TOC">back to top</a><br />
By default a relative abundance cut off of 0.005 (0.5%) is done to clear out low abundant clusters are most likely due to current technologies and PCR limitation and shouldnâ€™t be done in the experiment. If youâ€™re data is especially noisy or you want to be more stringent in your cut off use the <code>--fracCutOff</code> flag, for example the filter off all clusters at 1% and lower use the below example</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --fracCutOff 0.01</a></code></pre></div>
</section>
<section id="removing-singletons" class="level2">
<h2>Removing singletons</h2>
<p><a href="#TOC">back to top</a><br />
By default all singletons clusters are thrown out before analysis is done per replicate. To change this behavior use the <code>--clusterCutOff</code> flag</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="co">#to keep singletons </span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --clusterCutOff 0</a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#to remove clusters of size 2 and less </span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --clusterCutOff 2</a></code></pre></div>
</section>
<section id="marking-chimeras-and-checking-chimeras" class="level2">
<h2>Marking Chimeras and checking Chimeras</h2>
<p><a href="#TOC">back to top</a><br />
By default all sample clusters that are made up of mostly clusters with the CHI_ flag are removed from analysis. To keep chimeras use the <code>--keepChimera</code> flag. Also marking clusters can be done with <a href="qluster_usage.html#marking-possible-chimeric-sequence">SeekDeep qluster</a> but can also be done by processClusters by using the same flags as qluster to make them, <code>--markChimeras</code> and <code>--parFreqs</code> see <a href="qluster_usage.html#marking-possible-chimeric-sequence">SeekDeep qluster</a> for an explanation of these flags.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="co">#keep chimeras </span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --keepChimeras</a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">#mark chimeras when they have parents that are greater than 3x in frequency  </span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --makrChimeras --parFreqs 3</a></code></pre></div>
<p>Also marking all sequences that appear as chimeric can be removing low level recombinants. To try to prevent some of this from happening you can use the <code>--investigateChimeras</code> flag which will take clusters marked as chimeric and see if they are ever one of the two major clusters in an other sample. The reasoning behind this is that chimeric clusters need two parents and if a cluster appears as one of the two major clusters in a sample itâ€™s not possible to have two parents that are greater than it is.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --par pars.txt  --investigateChimeras</a></code></pre></div>
</section>
<section id="extra-cleanup" class="level2">
<h2>Extra cleanup</h2>
<p><a href="#TOC">back to top</a><br />
<code>SeekDeep processClusters</code> also offers the clean up of low frequencies one-off haplotypes at the end of clustering. This can be especially helpful for when there are no replicates to help correct for PCR noise and there might be several haplotypes that are at low frequency and only 1 different from a major higher in frequency haplotypes (e.g.Â major haplotype at 95% and a haplotype 1 base different at 1%). This can be a good alternative to simply collapsing all one-off haplotypes that would collapse haplotypes both at high frequency like 50% and 50%.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --strictErrors --pop-noErrors --collapseLowFreqOneOffs</a></code></pre></div>
<p>The different in frequencies is set by default so that the higher frequency haplotype has to be found at 30x the frequency of the lower frequency haplotypes (this number was arrived at by empirical means by looking at various control mixtures). This number can be changed using <code>--lowFreqMultiplier</code></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="ex">SeekDeep</span> processClusters --fastq output.fastq --strictErrors --pop-noErrors --collapseLowFreqOneOffs --lowFreqMultiplier 40</a></code></pre></div>
</section>
</section>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
