<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Nicholas Hathaway" />


<title>Multiplex SeekDeep Tutorial</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SeekDeep</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-sitemap"></span>
     
    Installing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="installingSeekDeep_Mac.html">
        <span class="fa fa-apple"></span>
         
        Mac OS
      </a>
    </li>
    <li>
      <a href="installingSeekDeep_Ubuntu.html">
        <span class="fa fa-linux"></span>
         
        Ubuntu
      </a>
    </li>
    <li>
      <a href="installingSeekDeep_Windows.html">
        <span class="fa fa-windows"></span>
         
        Windows
      </a>
    </li>
    <li>
      <a href="installingSeekDeep_vagrant.html">Vagrant/virtual image (any system)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Code
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://github.com/bailey-lab/SeekDeep">
        <span class="fa fa-github"></span>
         
        Github
      </a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-wrench"></span>
     
    Usages
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Pipeline</li>
    <li>
      <a href="extractor_usage.html">extractor/extractorPairedEnd</a>
    </li>
    <li>
      <a href="makeSampleDirectories_usage.html">makeSampleDirectories</a>
    </li>
    <li>
      <a href="qluster_usage.html">qluster</a>
    </li>
    <li>
      <a href="processClusters_usage.html">processClusters</a>
    </li>
    <li>
      <a href="popClusteringViewer_usage.html">popClusteringViewer</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Pipeline Wrapper</li>
    <li class="divider"></li>
    <li class="dropdown-header">Utilities</li>
    <li>
      <a href="genTargetInfoFromGenomes_usage.html">genTargetInfoFromGenomes</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-terminal"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Single End (454/IonTorrent/Stitched Illumina)</li>
    <li>
      <a href="tutorial_IonTorrent_MIDs.html">Ion Torrent with MIDs</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Illumina Paired End</li>
    <li>
      <a href="tutorial_PairedEnd_noMIDs.html">Paired End No MIDs</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc Info
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="illumina_paired_info.html">Illumina Paired End Info</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Multiplex SeekDeep Tutorial</h1>
<h4 class="author"><em>Nicholas Hathaway</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#installing-seekdeep">Installing SeekDeep</a></li>
<li><a href="#explanation-of-tutorial-data">Explanation of Tutorial Data</a></li>
<li><a href="#downloading-tutorial-data">Downloading Tutorial Data</a></li>
<li><a href="#files-created-to-run-pipeline">Files created to run Pipeline</a></li>
<li><a href="#setting-up-main-analysis-folder">Setting up main analysis folder</a><ul>
<li><a href="#running-seekdeep-setuptarampanalysis">Running SeekDeep setupTarAmpAnalysis</a></li>
<li><a href="#the-input-files">The input files</a><ul>
<li><a href="#all-the-input-files-for-tutorial">All the input files for tutorial</a></li>
<li><a href="#samples-names-file-to-id-file">Samples Names file to id file</a></li>
<li><a href="#samples-names-file-to-raw-data-input">Samples Names file to raw data input</a></li>
<li><a href="#samples-names-file-to-groupings-meta-data-file">Samples Names file to groupings meta data file</a></li>
<li><a href="#target-names-in-input-file-to-sample-names-and-id-files">Target Names in input file to sample names and id files</a></li>
</ul></li>
</ul></li>
<li><a href="#tutorial-example-data">Tutorial example data</a></li>
<li><a href="#running-tutorial-data">Running Tutorial Data</a><ul>
<li><a href="#required-options">Required options</a></li>
<li><a href="#optional-arguments">Optional arguments</a></li>
<li><a href="#passing-on-additional-arguments-to-the-default-arguments-of-the-3-main-sub-commands">Passing on additional arguments to the default arguments of the 3 main sub-commands</a></li>
</ul></li>
</ul>
</div>

<section id="installing-seekdeep" class="level1">
<h1>Installing SeekDeep</h1>
<p>To install SeekDeep check out the <a href="installingSeekDeep.html">install directions page</a></p>
</section>
<section id="explanation-of-tutorial-data" class="level1">
<h1>Explanation of Tutorial Data</h1>
<p>This is a made up dataset of targeted sequencing from the apical membrane antigen 1 (AMA1) protein from <em>Plasmodium Falciparum</em>. The dataset consists of 12 patients and PCR replicates for each one so a total 24 barcoded samples. These samples were pooled into three separate pools reusing the same barcodes and running on three different Ion Torrent runs. The data is split into two theoretical populations, patients from a high malarial transmission zone where malarial population should dense and there should be high complexity of infection (COI) compared to patients from a low malarial transmission zone.</p>
</section>
<section id="downloading-tutorial-data" class="level1">
<h1>Downloading Tutorial Data</h1>
<p><a href="#TOC">back to top</a><br />
Download the tutorial data materials using the below commands. This contains the example Ion Torrent data and the files created for this dataset.<br />
It also contains two scripts that contain all the below commands required to run the whole tutorial, it assumes <a href="">SeekDeep</a> has been compiled and is in your path (export PATH=/home/user/SeekDeep/bin/:$PATH)</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="fu">wget</span> http://baileylab.umassmed.edu/data/SeekDeepTutorialData/ver2_6_0/Multiplex_IT_Tutorial_Materials.tar.gz</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="fu">tar</span> -zxvf Multiplex_IT_Tutorial_Materials.tar.gz </a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="bu">cd</span> Multiplex_IT_Tutorial_Materials/</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">  </a></code></pre></div>
</section>
<section id="files-created-to-run-pipeline" class="level1">
<h1>Files created to run Pipeline</h1>
<p><a href="#TOC">back to top</a><br />
The Multiplex_IT_Tutorial_Materials directory contains several files that were created in order to run the <strong>SeekDeep</strong> pipeline and which are normally created each time a new dataset is sent through the pipeline.</p>
<ul>
<li><strong>pfama1_ids.tab.txt</strong>: This file supplies the forward and reverse primer along with the MIDs and their barcodes used in the experiment, see the <a href="extractor_usage.html#format-of-id-file">SeekDeep extractor</a> usage page for details on how this file should be formatted<br />
</li>
<li><strong>sampleNames.tab.txt</strong>: This file supplies how the samples in the experiment are organized across the different sequencing files and what MIDs were used for the samples including all MIDs if there are PCR replicates see <a href="makeSampleDirectories_usage.html#format-of-set-up-file">SeekDeep makeSampleDirectories</a> usage page for details on how this file is formatted. In this tutorial there are 24 samples for 12 patients spread across the three sequencing files.<br />
</li>
<li><strong>targetsToIndex.tab.txt</strong>: This is a file with two columns that lists what files contain which targets from the id file. One column is <code>index</code> and the other is <code>targets</code>, the names of the columns is important and must be these two. The <code>index</code> column has the name of the input files. The <code>targets</code> column has comma separated values indicating which targets from the id file are in that file. In this tutorial there is only one target, PFAMA1.<br />
</li>
<li><strong>groupingsFile.tab.txt</strong>: This file is useful for splitting up the final population clustering results into different groups if they are available. See <a href="processClusters_usage.html#supplying-grouping-info">SeekDeep processClusters</a> usage page for how this file should be set up. In this tutorial there is only one grouping, high vs low transmission zone so there is only one additional column to the sample column.</li>
</ul>
</section>
<section id="setting-up-main-analysis-folder" class="level1">
<h1>Setting up main analysis folder</h1>
<p><a href="#TOC">back to top</a><br />
Though all the tools of <code>SeekDeep</code> can be ran on it’s own, the tool <code>SeekDeep setupTarAmpAnalysis</code> can be used to set up the analysis and create all the commands with default settings.</p>
<section id="running-seekdeep-setuptarampanalysis" class="level2">
<h2>Running SeekDeep setupTarAmpAnalysis</h2>
<p><a href="#TOC">back to top</a><br />
To run <code>SeekDeep setupTarAmpAnalysis</code> at most two extra files have to be created to describe your input data.</p>
<p>An id file (here pfama1_ids.tab.txt) that describes the primers and MIDs, if still present, that were used in the experiment and a samples names file that indicates what samples are in which input files and if these samples have multiple replicates or only 1 run.</p>
<p>There are two ways this sample file can organized, one is by targets where each input file is already either de-multiplex or was only 1 sample to begin with, or, as in the case of this tutorial it can be organized by input file where MIDs are still present. When MIDs are still present 1 additional file is needed, targetToIndex.tab.txt, which is a file describing which targets are in the input files.</p>
<p>You can also supply a meta data file that aids in organizing data downstream.</p>
<p>See below for a visual depiction of how these files are set up.</p>
</section>
<section id="the-input-files" class="level2">
<h2>The input files</h2>
<p><a href="#TOC">back to top</a></p>
<section id="all-the-input-files-for-tutorial" class="level3">
<h3>All the input files for tutorial</h3>
<p>See below to see how they relate<br />
<img src="images/byIndex_tutorial_files.jpg" /></p>
</section>
<section id="samples-names-file-to-id-file" class="level3">
<h3>Samples Names file to id file</h3>
<p><img src="images/byIndex_Id_to_samples_names.jpg" /></p>
</section>
<section id="samples-names-file-to-raw-data-input" class="level3">
<h3>Samples Names file to raw data input</h3>
<p><img src="images/byIndex_sampleNames_to_rawData.jpg" /></p>
</section>
<section id="samples-names-file-to-groupings-meta-data-file" class="level3">
<h3>Samples Names file to groupings meta data file</h3>
<p>If you have meta data you would like to associate with your samples you can give a file adding the info. This info will be given with the final results from the SeekDeep pipeline to make downstream analysis easier by having it already in the tables. SeekDeep will also organize data separated out by meta data to ease viewing of the data by group. See the below diagram as how this meta data should be matched up with the previous files described in this section. The name of the group metadata file is given to the <code>--groupMeta</code> argument.<br />
<img src="images/byIndex_sampleNames_to_groupings.jpg" /></p>
</section>
<section id="target-names-in-input-file-to-sample-names-and-id-files" class="level3">
<h3>Target Names in input file to sample names and id files</h3>
<p>To indicate which targets are in which input files (some set up have different set of targets in each input file) you can supply a targets to index file to do so, see below for how this relates to the id and sample names file. The name of this file is given to the <code>--targetsToIndex</code> flag.<br />
<img src="images/byIndex_targetsToIdFile.jpg" /></p>
</section>
</section>
</section>
<section id="tutorial-example-data" class="level1">
<h1>Tutorial example data</h1>
<p>The tutorial data is a made up experiment where an investigation is being done for investigating parasite diversity in 12 patients from two different regions, 6 from 1 region and the other 6 from a second region. Meta data is provided for each patient so results per region can be viewed separately.</p>
<p>A theoretical display of the regions for the experiment explained above.<br />
<img src="images/Regions.jpg" alt="Regions" /></p>
</section>
<section id="running-tutorial-data" class="level1">
<h1>Running Tutorial Data</h1>
<p><a href="#TOC">back to top</a></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="ex">SeekDeep</span> setupTarAmpAnalysis --technology iontorrent --inputDir rawDirectory/ --inputFilePat <span class="st">&quot;.*.fastq.gz&quot;</span> --samples sampleNames.tab.txt --idFile pfama1_ids.tab.txt --groupMeta groupingsFile.tab.txt --outDir analysis --byIndex --targetsToIndex targetsToIndex.tab.txt <span class="st">&quot;--extraExtractorCmds=--checkRevComplementForMids --qualWindow 50,5,18&quot;</span> --numThreads 7</a></code></pre></div>
<section id="required-options" class="level2">
<h2>Required options</h2>
<ul>
<li><strong>--samples</strong> - The file with samples names and raw data file names as explained above<br />
</li>
<li><strong>--outDir</strong> - An output directory where analysis will be set up<br />
</li>
<li><strong>--inputDir</strong> - The input raw data directory<br />
</li>
<li><strong>--idFile</strong> - The id file explained above<br />
</li>
<li><strong>--byIndex</strong> - Indicates that the <code>--samples</code> file’s first column is file names and not by target<br />
</li>
<li><strong>--targetsToIndex</strong> - A file that has what targets are in what files from the <code>--samples</code> file, (if all targets are present in all files then this doesn’t have to be supplied)</li>
</ul>
</section>
<section id="optional-arguments" class="level2">
<h2>Optional arguments</h2>
<ul>
<li><strong>--technology</strong> - This sets the technology to indicate what the raw data is, this can be 454, Illumina(default), or IonTorrent.<br />
</li>
<li><strong>--inputFilePat</strong> - This is a regex pattern to match files in the raw data input directory to use in the analysis, when the <code>--technology</code> is set to 454 or ion torrent this defaults to <code>&quot;.*.fastq&quot;</code> which would be all files ending with fastq, since in this case the input is gzipped files it is being set here to <code>&quot;.*.fastq.gz&quot;</code> which will automatically be unzipped.<br />
</li>
<li><strong>--groupMeta</strong> - A file with meta data to associate with the input samples, see above to see how this file should be associated with the other input files<br />
</li>
<li><strong>--numThreads</strong> - The number of CPUs to be utilized to speed up analysis</li>
</ul>
</section>
<section id="passing-on-additional-arguments-to-the-default-arguments-of-the-3-main-sub-commands" class="level2">
<h2>Passing on additional arguments to the default arguments of the 3 main sub-commands</h2>
<p>Default scripts are created for each of the downstream analysis commands and additional arguments can be passed onto these scripts via the following three commands.<br />
* <strong>--extraExtractorCmds</strong> - Any extra commands to append to the default ones for the extractor step, should be given in quotes e.g. <code>--extraExtractorCmds=&quot;--checkRevComplementForMids --qualWindow 50,5,18&quot;</code> * <strong>--extraQlusterCmds</strong> - Any extra commands to append to the default ones for the qluster step, should be given in quotes<br />
* <strong>--extraProcessClusterCmds</strong> - Any extra commands to append to the default ones for the processClusters step, should be given in quotes</p>
<p>This will extract the raw data from the input directory and unzip if necessary. Also id files will also be copied into the directory as well. Also default scripts will be created that will run the rest of the analysis, with defaults for the technology given to <code>--technology</code>, all of which can be ran by the file, runAnalysis.sh in the output directory or one at a time.</p>
<p><strong>./runAnalysis.sh</strong></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#!/usr/bin/env bash</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">##run all parts of the pipeline</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="va">numThreads=</span>1</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="kw">if [[</span> <span class="va">$#</span> <span class="ot">-eq</span> 1<span class="kw"> ]]</span>; <span class="kw">then</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">    <span class="va">numThreads=$1</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="kw">fi</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="ex">SeekDeep</span> runMultipleCommands --cmdFile extractorCmds.txt      --numThreads <span class="va">$numThreads</span> --raw</a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="ex">SeekDeep</span> runMultipleCommands --cmdFile qlusterCmds.txt        --numThreads <span class="va">$numThreads</span> --raw</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="ex">SeekDeep</span> runMultipleCommands --cmdFile processClusterCmds.txt --numThreads <span class="va">$numThreads</span> --raw</a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="ex">SeekDeep</span> runMultipleCommands --cmdFile genConfigCmds.txt      --numThreads <span class="va">$numThreads</span> --raw</a></code></pre></div>
<p>The command <code>SeekDeep runMultipleCommands</code> is an utility in <code>SeekDeep</code> that allows the running of multiple commands and gives an output file in json for how each command executed. See the usage pages for each command to see an explanation of the flags used in the analysis.</p>
<p>The files extractorCmds.txt, qlusterCmds.txt, processClusterCmds.txt, and genConfigCmds.txt contain commandline commands on each line to run the analysis. The <code>SeekDeep runMultipleCommands</code> is command in <code>SeekDeep</code> that can take in such a file and run them in parallel speeding up analysis.</p>
<p>See below to see how these command files match up to the pipeline.</p>
<figure>
<img src="images/pipeline.jpg" alt="SeekDeep Pipeline" /><figcaption>SeekDeep Pipeline</figcaption>
</figure>
<p>And then to start the server to explore the results interactively run the file <code>startServerCmd.sh</code> after running the above command files. <strong>./startServerCmd.sh</strong></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">#!/usr/bin/env bash</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co"># Will automatically run the server in the background and with nohup so it will keep running</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw">if [[</span> <span class="va">$#</span> <span class="ot">-ne</span> 2<span class="kw"> ]]</span> <span class="kw">&amp;&amp; [[</span> <span class="va">$#</span> <span class="ot">-ne</span> 0<span class="kw"> ]]</span>; <span class="kw">then</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">    <span class="bu">echo</span> <span class="st">&quot;Illegal number of parameters, needs either 0 or 2 argument, if 2 args 1) port number to server on 2) the name to serve on&quot;</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">    <span class="bu">echo</span> <span class="st">&quot;Examples&quot;</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6">    <span class="bu">echo</span> <span class="st">&quot;./startServerCmd.sh&quot;</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7">    <span class="bu">echo</span> <span class="st">&quot;./startServerCmd.sh 9882 pcv2&quot;</span>    </a>
<a class="sourceLine" id="cb4-8" data-line-number="8">    <span class="bu">exit</span>    </a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="kw">fi</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="kw">if [[</span> <span class="va">$#</span> <span class="ot">-eq</span> 2<span class="kw"> ]]</span>; <span class="kw">then</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12">    <span class="fu">nohup</span> SeekDeep popClusteringViewer --verbose --configDir <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>/serverConfigs --port <span class="va">$1</span> --name <span class="va">$2</span> <span class="kw">&amp;</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="kw">else</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14">    <span class="fu">nohup</span> SeekDeep popClusteringViewer --verbose --configDir <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>/serverConfigs <span class="kw">&amp;</span> </a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="kw">fi</span></a></code></pre></div>
<p>Check out <a href="http://baileylab.umassmed.edu/SeekDeepExample1" class="uri">http://baileylab.umassmed.edu/SeekDeepExample1</a> to see an example of the server hosting of the tutorial data by <code>SeekDeep popClusteringViewer</code>.</p>
</section>
</section>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
